packages/next/src/server/lib/router-server.ts

```diff
@@ -473,7 +474,11 @@ export async function initialize(opts: {
           !res.getHeader('cache-control') &&
           matchedOutput.type === 'nextStaticFolder'
         ) {
-          if (opts.dev && !isNextFont(parsedUrl.pathname)) {
+          if (
+            opts.dev &&
+            !isNextFont(parsedUrl.pathname) &&
+            !isMetadataStaticFile(parsedUrl.pathname || '')
```

Comment:
It's better to keep no-cache in dev. For instance if the icon file is updated then the previous icon might still be cached in the browser.